dev/src/reference.ts

```diff
@@ -2177,9 +2177,18 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
                 // query cursor. Note that we do not use backoff here. The call to
                 // `requestStream()` will backoff should the restart fail before
                 // delivering any results.
-                request = this.startAfter(lastReceivedDocument).toProto(
-                  lastReceivedDocument.readTime
-                );
+                if (this._queryOptions.kindless) {
```

Comment:
I originally wanted to abstract away the recursive delete requirements from the Query class, but I can't think of another way to omit readTime elegantly. At some point, should I just rename the queryOptions to include recursive delete in the naming instead of "kindless"?

Reply:
Can we just be explicit here and create a new setting that either mentions read time directly or addresses the concept that we want to change? Something like `enforceConsistency` or `requiresConsistency` might work.

Reply:
That makes sense. Changed to `requireConsistency`.